### ISSUE TITLE
Missed gestures

### ISSUE THREAD

**Issue by pzac** [Robbendebiene/Gesturefy#622]
Hi, I've been recently noticing that some gestures are missed: the overlay with the intended action appears, but the action is not executed. This seems to happen especially when it's a fast, short gesture (like page back/forward), regardless of how many times I repeat. If I try again, moving very slowly, the gesture works again. Any idea?

My setup: MacOS 12.0.1, FF 94.0.1, GF 3.2.3.

Thanks for your work on this extension!

**Comment by Robbendebiene** [comment]
Hi, can you please share your config? (you can export it in the settings page under "about").

Honestly the only explanation I could come up with is that shortly before you complete the gesture you somehow make a move that leads to an unmatched gesture.

When did the problem first occur? Maybe after the update to Firefox 94? Does it work with a previous Gesturefy version, like 3.2.1 ?

Any special devices things like a high resolution display or high dpi mouse?

Since I can't reproduce the problem I'm only guessing an rely on further investigation by you.

------------
Another idea:
What's new in version 3.2.2 is that you can now cancel the Gesture by clicking another mouse button. Might this be an option that you somehow trigger another mouse button?

**Comment by SteveOnorato** [comment]
Same problem for me in 3.2.3.  Quick short gestures hardly ever work now, but they had never failed in previous months of use.  Confirmed that rolling back to 3.2.1 works great, but 3.2.3 is an extremely frustrating experience.

FF 94.0, MacOS 11.6, MacBook Pro (16-inch 2019), 5K external monitor, Logitech M500s mouse.

**Comment by SteveOnorato** [comment]
Here's my current config.json.  I removed all but 1 gesture to debug.  Also tried all 3 values for "matchingAlgorithm"
```
{
  "Blacklist": [],
  "Gestures": [
    {
      "pattern": [
        [
          0,
          235
        ]
      ],
      "command": {
        "name": "CloseTab",
        "settings": {
          "nextFocus": "default",
          "closePinned": true
        }
      }
    }
  ],
  "Settings": {
    "Gesture": {
      "mouseButton": 2,
      "suppressionKey": "",
      "distanceThreshold": 10,
      "distanceSensitivity": 10,
      "Timeout": {
        "active": true,
        "duration": 3
      },
      "Trace": {
        "display": true,
        "Style": {
          "opacity": 0.8,
          "strokeStyle": "#00AAA0cc",
          "lineWidth": 10,
          "lineGrowth": true
        }
      },
      "Directions": {
        "display": false,
        "Style": {
          "color": "#FFFFFF",
          "backgroundColor": "#000000",
          "backgroundOpacity": 0.3,
          "fontSize": "8vh",
          "textAlign": "center"
        }
      },
      "Command": {
        "display": true,
        "Style": {
          "backgroundColor": "#0000004d",
          "backgroundOpacity": 0.3,
          "fontSize": "6vh",
          "fontColor": "#FFFFFFff"
        }
      },
      "matchingAlgorithm": "strict"
    },
    "Rocker": {
      "active": false,
      "leftMouseClick": {
        "name": "PageBack"
      },
      "rightMouseClick": {
        "name": "PageForth"
      }
    },
    "Wheel": {
      "active": false,
      "mouseButton": 1,
      "wheelUp": {
        "name": "FocusRightTab",
        "settings": {
          "cycling": true,
          "excludeDiscarded": false
        }
      },
      "wheelDown": {
        "name": "FocusLeftTab",
        "settings": {
          "cycling": true,
          "excludeDiscarded": false
        }
      }
    },
    "General": {
      "updateNotification": true,
      "theme": "light"
    }
  }
}
```

**Comment by pzac** [comment]
These are my settings:

```json
{
  "Gestures": [
    {
      "pattern": [
        [
          0,
          -1
        ]
      ],
      "command": {
        "name": "ScrollTop",
        "settings": {
          "duration": 100
        }
      }
    },
    {
      "pattern": [
        [
          1,
          0
        ]
      ],
      "command": {
        "name": "PageForth"
      }
    },
    {
      "pattern": [
        [
          -1,
          0
        ]
      ],
      "command": {
        "name": "PageBack"
      }
    },
    {
      "pattern": [
        [
          9,
          264
        ]
      ],
      "command": {
        "name": "CloseTab",
        "settings": {
          "nextFocus": "default",
          "closePinned": true
        }
      }
    }
  ],
  "Settings": {
    "Gesture": {
      "distanceThreshold": 8,
      "deviationTolerance": 0.15,
      "matchingAlgorithm": "combined"
    }
  }
}
```

**Comment by pzac** [comment]
I confirm that version 3.2.1 works perfectly. Additional hardware info, if helps: bluetooth Logitech M535 mouse, 5k external display.

**Comment by Robbendebiene** [comment]
@pzac @SteveOnorato Can you please reproduce the problem in the white area below: https://codepen.io/Robbendebiene/pen/JjyZraM and post the outputs here?

I'm pretty sure the following is causing the problem:

> What's new in version 3.2.2 is that you can now cancel the Gesture by clicking another mouse button.

My guess is that somehow Firefox on Mac reports a wrong mouse button. Maybe I can work around that.

**Comment by SteveOnorato** [comment]
This corresponds to a failed gesture: 
<img width="313" alt="image" src="https://user-images.githubusercontent.com/25069709/140998927-aeed2832-181c-4e4a-ac67-3ac56087812b.png">

If I go slower, there isn't a pointermove line with "buttons: 0": 
<img width="294" alt="image" src="https://user-images.githubusercontent.com/25069709/140999409-eb907e01-56ab-4e7f-ac79-113966bc7ead.png">

**Comment by Robbendebiene** [comment]
Thanks for the data! This is exactly as I suspected.

I've just implemented a workaround.
Can someone please verify that it solves the problem?

To do this:
1. Download latest Gesturefy master https://github.com/Robbendebiene/Gesturefy/archive/refs/heads/master.zip
2. Extract zip somewhere
3. Open `about:debugging` switch to "This Firefox" and click "Load temporary addon"
4. Select the `manifest.json` from the extracted zip inside the `src` directory
5. Test if the error is gone

**Comment by SteveOnorato** [comment]
Can't reproduce the problem with that build, so looks like you fixed it :-)

**Comment by pzac** [comment]
Yes, this works for me, too! 👍

**Comment by Robbendebiene** [comment]
Thanks to both of you for your quick feedback. I'm going to release a new version that contains the fix.

**Comment by Robbendebiene** [comment]
If any one of you finds the time it would be good to create a Firefox bug report on [bugzilla](https://bugzilla.mozilla.org/) linking to the [codepen](https://codepen.io/Robbendebiene/pen/JjyZraM) and the results you've got. Because the pointermove line with "buttons: 0" shouldn't occur in this case.

**Comment by Robbendebiene** [comment]
This should be fixed in [Gesturefy 3.2.4](https://github.com/Robbendebiene/Gesturefy/releases/tag/v3.2.4)
